src/main/java/com/google/sps/servlets/LoginServlet.java

```
@@ -0,0 +1,58 @@
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdToken.Payload;
+import com.google.api.client.googleapis.auth.oauth2.GoogleIdTokenVerifier;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonParser;
+
+import java.util.Collections;
+import com.google.api.client.extensions.appengine.http.UrlFetchTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+    UrlFetchTransport transport = new UrlFetchTransport();
+    JacksonFactory jacksonFactory = new JacksonFactory();
+    GoogleIdTokenVerifier verifier = new GoogleIdTokenVerifier.Builder(transport, jacksonFactory).setAudience(Collections.singletonList("758286654746-8e5tr4b5tr0gukbkdjpb6vj6upd9pl6l.apps.googleusercontent.com")).build();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        response.setContentType("text/html");
+        response.setHeader("Set-Cookie", "HttpOnly;Secure;SameSite=Strict");
+        String idTokenString = request.getParameter("id_token");
+        JsonObject json = new JsonObject();
+        try {
+            GoogleIdToken idToken = verifier.verify(idTokenString);
+            Payload payload = idToken.getPayload();
+            String userId = payload.getSubject();
+            json = new JsonParser().parse(payload.toString()).getAsJsonObject();
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
```

we should return the error back to the user. We should also catch the most specific checked exception(s) possible, since Exception catches everything, including runtime exceptions like NullPointerExceptions, which might be unrelated to the actual logic of logging in failing.

---

Changed to GeneralSecurityException and IOException only (from GoogleIdToken documentation)

---

we should add a TODO to make sure to propagate an error back to the user, if we're not doing that in this change